cmd/cody-gateway/internal/limiter/prefix.go

```diff
@@ -1,12 +1,30 @@
 package limiter
 
+import (
+	"fmt"
+	"github.com/sourcegraph/sourcegraph/internal/codygateway"
+)
+
 func NewPrefixRedisStore(prefix string, store RedisStore) RedisStore {
 	return &prefixRedisStore{
 		prefix: prefix,
 		store:  store,
 	}
 }
 
+// featurePrefix is the prefix used by redis store for the given
+// feature because we need to rate limit by feature.
+func featurePrefix(feature codygateway.Feature) string {
+	return fmt.Sprintf("%s:", feature)
+}
+
+func NewFeatureUsageStore(feature codygateway.Feature, store RedisStore) RedisStore {
```

Comment:
```suggestion
// NewFeatureUsageStore rate-limits per feature on top of the underlying store.
func NewFeatureUsageStore(feature codygateway.Feature, store RedisStore) RedisStore {
```

I think convention might dictate that `store` come first as well, but doesn't really matter